test/common.py

```diff
@@ -296,20 +312,31 @@ def remove_prefix(text, prefix):
             expected_file += "-" + subname
         expected_file += ".expect"
         expected = None
+        try:
+            with open(expected_file) as f:
+                expected = f.read()
+        except IOError as e:
+            if e.errno == errno.ENOENT:
+                if ACCEPT:
+                    print("Accepting output for {}:\n\n{}".format(munged_id, s))
+                    with open(expected_file, 'w') as f:
+                        f.write(s)
+                    return  # don't keep going
+                else:
+                    raise RuntimeError(
+                        ("I got this output for {}:\n\n{}\n\n"
+                         "No expect file exists; to accept the current output, run:\n"
+                         "python {} {} --accept").format(munged_id, s, __main__.__file__, munged_id))
+            else:
+                raise
         if ACCEPT:
-            with open(expected_file, 'w') as f:
-                f.write(s)
+            if expected == s:
```

Comment:
There's sth wrong with this code. Why do we have two places where we accept (overwrite) the file now? Opening for writing will create it anyway

Reply:
I'll factor the duplicate code into a helper method